wom/src/main/scala/wom/graph/expression/AnonymousExpressionNode.scala

```diff
@@ -14,7 +14,15 @@ object AnonymousExpressionNode {
                                                      expression: WomExpression,
                                                      inputMapping: Map[String, OutputPort],
                                                      constructor: AnonymousExpressionConstructor[T]): ErrorOr[T] = {
-    ExpressionNode.buildFromConstructor(constructor)(identifier, expression, inputMapping)
+    // Anonymous expression nodes are created and then immediately consumed.
+    // Name mangling prevents lookups from finding them and using them after their expiration date.
+    // Naturally, leading underscore is an illegal identifier so would never happen IRL.
+    val mangledIdentifier = identifier.copy(
```

Comment:
maybe `unfindable` rather than... `mangled`?

Reply:
Is the trouble with "mangled" that it sounds unpleasant, or that it is inaccurate?

I think it would be a fairly standard term for what's going on here - see e.g. https://en.wikipedia.org/wiki/Name_mangling#Creating_unique_names_for_inner_and_anonymous_classes

Reply:
Huh, TIL...

In my mental models "mangle" would involve changing it in a way that makes it look totally unrecognizable (rather than just adding a small prefix, for example), but I agree - according to that link, `mangle` is indeed the correct term for what's happening here